packages/stream_chat/lib/src/client/channel.dart

```diff
@@ -2495,6 +2497,28 @@ class ChannelClientState {
     }));
   }
 
+  // Logic taken from the backend SDK
+  // https://github.com/GetStream/chat/blob/9245c2b3f7e679267d57ee510c60e93de051cb8e/types/channel.go#L1136-L1150
+  bool _shouldUpdateChannelLastMessageAt(Message message) {
+    if (message.shadowed) return false;
+    if (message.isEphemeral) return false;
+
+    final config = channelState.channel?.config;
+    if (message.isSystem && config?.skipLastMsgUpdateForSystemMsgs == true) {
+      return false;
+    }
+
+    final currentUser = _channel._client.state.currentUser;
+    final restrictedVisibility = message.restrictedVisibility;
+    if (restrictedVisibility case final visibility?) {
+      if (visibility.isNotEmpty && !visibility.contains(currentUser?.id)) {
```

Comment:
```suggestion
      if (visibility.isNotEmpty && visibility.contains(currentUser?.id)) {
```

Reply:
We are returning false if the message is not supposed to be for the current user. The `restricted_visibility` list contains the user-ids of the users who should receive this message.